uspace/app/bdsh/cmds/modules/ls/ls.c

```diff
@@ -87,14 +95,52 @@ static unsigned int ls_start(ls_job_t *ls)
  *
  * @param de		Directory element.
  */
-static void ls_print(struct dir_elem_t *de)
+static errno_t ls_print(struct dir_elem_t *de)
 {
-	if (de->s.is_file)
-		printf("%-40s\t%llu\n", de->name, (long long) de->s.size);
-	else if (de->s.is_directory)
-		printf("%-40s\t<dir>\n", de->name);
+	int width = 13;
+
+	if (de->s.is_file) {
+		if (ls.exact_size) {
+			printf("%-40s\t%*llu\n", de->name, width, (long long) de->s.size);
```

Comment:
Also, early return here would avoid another level of indentation.

Reply:
I dont understand what you mean. Every case needs to be checked -> there is no way to "return early"

Reply:
After this `printf()`, there is no other code present until the end of function. Only `else` branches. Therefore, you can put `return EOK;` right after the `printf()`. When you do that, you don't need `else` after it, so you can reduce indentation of the block of code below.

Horizontal space is a premium currency, so any opportunity to reduce the amount of indentation used is a good thing.